lib/config/config.js

```diff
@@ -144,6 +320,13 @@ class Config {
 	 */
 	#processorName;
 
+	/**
+	 * A collection of compiled validators for rules that have already
+	 * been validated.
+	 * @type {WeakMap}
+	 */
+	#validators = new WeakMap();
```

Comment:
I think this cache should be at the module level, i.e., a single cache for the whole process.

For comparison, when linting this project (`node bin/eslint .`), `ajv.compile()` is called 257 times before this change, but  2880 times after this change.

Reply:
Before this change, this cache was an instance property of `RuleValidator`. But the `config.js` module was using a single instance of `RuleValidator`, so there were no duplicate validators created (except for inline config in `Linter`, which was unnecessary).

https://github.com/eslint/eslint/blob/8ed32734cc22988173f99fd0703d50f94c60feb8/lib/config/config.js#L15-L22

Reply:
eslintrc also keeps this cache at the module level: https://github.com/eslint/eslintrc/blob/556e80029f01d07758ab1f5801bc9421bca4b072/lib/shared/config-validator.js#L28

Reply:
Nice catch. I'll update that.